src/utils/multipart-parser.js

```diff
@@ -0,0 +1,440 @@
+import {File} from 'fetch-blob/from.js';
+import {FormData} from 'formdata-polyfill/esm.min.js';
+
+let s = 0;
+const S = {
+	PARSER_UNINITIALIZED: s++,
+	START: s++,
+	START_BOUNDARY: s++,
+	HEADER_FIELD_START: s++,
+	HEADER_FIELD: s++,
+	HEADER_VALUE_START: s++,
+	HEADER_VALUE: s++,
+	HEADER_VALUE_ALMOST_DONE: s++,
+	HEADERS_ALMOST_DONE: s++,
+	PART_DATA_START: s++,
+	PART_DATA: s++,
+	PART_END: s++,
+	END: s++
+};
+
+let f = 1;
+const F = {
+	PART_BOUNDARY: f,
+	LAST_BOUNDARY: f *= 2
+};
+
+const LF = 10;
+const CR = 13;
+const SPACE = 32;
+const HYPHEN = 45;
+const COLON = 58;
+const A = 97;
+const Z = 122;
+
+const lower = function (c) {
+	return c | 0x20;
+};
+
+class MultipartParser {
+	constructor(string) {
+		this.index = null;
+		this.flags = 0;
+
+		this.boundaryChars = {};
+
+		string = '\r\n--' + string;
+		const ui8a = new Uint8Array(string.length);
+		for (let i = 0; i < string.length; i++) {
+			ui8a[i] = string.charCodeAt(i);
+			this.boundaryChars[ui8a[i]] = true;
+		}
+
+		this.boundary = ui8a;
+		this.lookbehind = new Uint8Array(this.boundary.length + 8);
+		this.state = S.START;
+	}
+
+	write(ui8a) {
+		let i = 0;
+		const length_ = ui8a.length;
+		let previousIndex = this.index;
+		let {lookbehind, boundary, boundaryChars, index, state, flags} = this;
+		const boundaryLength = this.boundary.length;
+		const boundaryEnd = boundaryLength - 1;
+		const bufferLength = ui8a.length;
+		let c;
+		let cl;
+
+		const mark = name => {
+			this[name + 'Mark'] = i;
+		};
+
+		const clear = name => {
+			delete this[name + 'Mark'];
+		};
+
+		const callback = (name, start, end, ui8a) => {
+			if (start !== undefined && start === end) {
+				return;
+			}
+
+			const callbackSymbol = 'on' + name.slice(0, 1).toUpperCase() + name.slice(1);
+			if (callbackSymbol in this) {
+				this[callbackSymbol](ui8a && ui8a.subarray(start, end));
+			}
+		};
+
+		const dataCallback = (name, clear) => {
+			const markSymbol = name + 'Mark';
+			if (!(markSymbol in this)) {
+				return;
+			}
+
+			if (clear) {
+				callback(name, this[markSymbol], i, ui8a);
+				delete this[markSymbol];
+			} else {
+				callback(name, this[markSymbol], ui8a.length, ui8a);
+				this[markSymbol] = 0;
+			}
+		};
+
+		for (i = 0; i < length_; i++) {
+			c = ui8a[i];
+
+			switch (state) {
+				case S.PARSER_UNINITIALIZED:
+					return i;
```

Comment:
The `state` variable is never set to `PARSER_UNINITIALIZED`, in fact `PARSER_UNINITIALIZED` isn't used at all so I think that we should remove it

```suggestion
```